api/serviceEntries/serviceEntriesModel.js

```diff
@@ -1,107 +1,99 @@
-  //   'service_types.service_type_id',
-  //   'locations.location_id'
-  // );
+  return await knex('service_entries as se')
+    .join('providers as pv', 'se.primary_provider_id', 'pv.provider_id')
+    .join('recipients as r', 'se.primary_recipient_id', 'r.recipient_id')
+    .join(
+      'service_type_programs as stp',
+      'se.service_type_program_id',
+      'stp.service_type_program_id'
+    )
+    .join('service_types as st', 'st.service_type_id', 'stp.service_type_id')
+    .join('programs as pg', 'pg.program_id', 'stp.program_id')
+    .join('statuses as sta', 'sta.status_id', 'se.status_id')
+    .join('locations as l', 'l.location_id', 'se.location_id')
+    .select(
+      'service_entry_id',
+      'provider_first_name',
+      'provider_last_name',
+      'recipient_first_name',
+      'recipient_last_name',
+      'program_name',
+      'service_type_name',
+      'apply_service_to_household',
+      'service_date',
+      'service_time',
+      'service_duration',
+      'service_value',
+      'service_quantity',
+      'service_entry_notes',
+      'service_entry_data',
+      'service_unit_id',
+      'status',
+      'service_rating_id',
+      'l.location_id',
+      'address',
+      'se.created_at',
+      'se.updated_at'
+    )
+    .where('service_entry_id', id)
+    .first();
 };
+
 const createServiceEntry = async (newServiceEntry) => {
   return await knex('service_entries').insert(newServiceEntry, ['*']);
-
-  //   let newServiceEntriesId;
-  //   try {
-  //     await knex.transaction(async (trx) => {
-  //       const createdServiceEntries = await trx('service_entries')
-  //         .insert([
-  //           { service_type_id: serviceEntries.service_type_id },
-  //           { location_id: serviceEntries.location_id },
-  //           { service_time: serviceEntries.service_time },
-  //           { service_date: serviceEntries.service_date },
-  //           { service_entry_data: serviceEntries.service_entry },
-  //         ])
-  //         .returning('*');
-
-  //       newServiceEntriesId = createdServiceEntries[0].service_entry_id;
-
-  //       // if this is all meant to let a single service entry have multiple providers and recipients, why isn't it mapping??
-
-  //       await trx('service_entry_providers').insert([
-  //         { service_entry_id: newServiceEntriesId },
-  //         { provider_id: serviceEntries.provider_id },
-  //       ]);
-  //       await trx('service_entry_recipients').insert([
-  //         { service_entry_id: newServiceEntriesId },
-  //         { recipient_id: serviceEntries.recipient_id },
-  //       ]);
-  //     });
-
-  //     return await findById(newServiceEntriesId);
-  //   } catch (err) {
-  //     throw new Error(err);
-  //   }
 };
 
 // Can only update service_entries data in table on front end
 // Can't update "extra" data that's tacked onto the responses
 // Can't edit: first_name, last_name, service_type_name, status_name _
```

Comment:
I think these comments apply to the previous DB, but I don't see any reason why these fields can't be updated now that we've simplified the structure?